pf/internal/plugin/provider_server.go

```diff
@@ -36,10 +38,18 @@ type providerServer struct {
 	provider      ProviderWithContext
 	keepSecrets   bool
 	keepResources bool
+
+	configEncoding tfbridge.ConfigEncoding
 }
 
-func NewProviderServerWithContext(provider ProviderWithContext) pulumirpc.ResourceProviderServer {
-	return &providerServer{provider: provider}
+func NewProviderServerWithContext(
```

Comment:
Once we started modifying this plumbing it's tempting to modify further; installed config encoding here.